src/include/platform/OTARequestorStorage.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+class OTARequestorStorage
+{
+public:
+    using ProviderLocationType = app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type;
+
+    virtual ~OTARequestorStorage() {}
+
+    virtual CHIP_ERROR StoreProviderLocation(const ProviderLocationType & provider) = 0;
```

Comment:
Need an iterator over all default provider locations to support the cluster. See GroupDataProvider.h for patterns.

Overall, you need to be able to use this storage backing API to:

- Iterate over default providers (to assist in implementing fabric-scoped/fabric-filtered DefaultOTAProviders list in cluster server, and iteration by application over candidates)
- Add a default provider for a given fabric scope at a given global list index
- Remove a default provider for a given fabric scope at a given global list index
- Update a default provider for a given fabric scope at a given global list index
- Determine if there is an OTA in progress and its update token
  - Looks like there is no metadata being loaded/stored, which may be OK, but consider allowing storage of fabric index of OTA provider used, Node ID of last OTA provider used, last written storage index and the last OTA URL, for ongoing OTA transfer
- Clear all OTA data related to an OTA previously in progress

Reply:
Ok, I guess I need to add a comment to make it clearer. Here I mean the current provider from which a new image has just been downloaded, not the list of default providers.

Reply:
Why wouldn't all non-volatile storage for the feature be managed by a single interface and integration point? Is the default OTA provider list not stored yet?

Reply:
It just wanted to cover the part that I'm assigned to for now. That is, automatic UpdateNotifyApplied. But yes, I expect that serializing the default OTA provider list will be added to this interface, too.

Reply:
Added methods to store the default OTA Provider list.